### PR TITLE
Use a default schema provider without path in case none is supplied

### PR DESCRIPTION
**Pull request by laeubi**
Even though no search path is provided we need to decode the URL and see if we can find the schema in the current workspace.

Also fixes an issue where schema can't be opened by double click on linux.

Fix https://github.com/eclipse-pde/eclipse.pde/issues/890

### CODE REVIEW

**Comment by laeubi** [comment]
@vik-chand @HannesWell please give +1 to merge this bugfix/regression regardless of M3 freeze.

**Comment by iloveeclipse** [comment]
> please give +1 to merge this bugfix/regression regardless of M3 freeze.

Fixes for regressions do not need approval in M3

**Review by merks — APPROVED** [review]
Don’t wait for reviews.

**Comment by laeubi** [comment]
Okay then lets merge as soon as CI checks have completed.

**Review by iloveeclipse — APPROVED** [review]

**Comment by vik-chand** [comment]
thanks @laeubi  @iloveeclipse

**Pull request merged**
